pyQuARC/code/custom_checker.py

```diff
@@ -32,19 +37,31 @@ def _get_path_value_recursively(subset_of_metadata_content, path_list, container
             container.append(subset_of_metadata_content)
             return
         new_path = path_list[1:]
-        if isinstance(root_content, str) or isinstance(root_content, int):
+        if isinstance(root_content, str) or isinstance(root_content, int) or isinstance(root_content, float):
             container.append(root_content)
+            return
         elif isinstance(root_content, list):
+            if not new_path:
+                container.append(root_content)
+                return
+            if len(new_path) == 1 and query_params:
+                try:
+                    root_content = next((x for x in root_content if x[query_params[0]] == query_params[1]))
+                    root_content = root_content[new_path[0]]
+                    container.append(root_content)
+                except:
+                    container.append(None)
+                return
```

Comment:
This logic seems a little too complex. Can we refactor this a little?

Reply:
HALP